shared/lib_battery_lifetime.cpp

```diff
@@ -78,6 +80,16 @@ lifetime_state::lifetime_state(const std::shared_ptr<lifetime_nmc_state>& nmc) {
     q_relative = fmin(nmc->q_relative_li, nmc->q_relative_neg);
 }
 
+lifetime_state::lifetime_state(const std::shared_ptr<lifetime_lmolto_state> &lmolto) {
```

Comment:
The idea behind this set of constructors where the inputs are a single component state class (for the calcyc, nmc, or the lmo/lto) doesn't make sense, thanks for bringing my attention to it. I've deleted those 3 constructors since they weren't even being used!